### PR TITLE
Add a general `sendM` combinator.

### PR DESCRIPTION
**Pull request by patrickt**
We could also call this `sendM` if we wanted to avoid a conflict with `mtl`, but given that the effect is called `Lift` I think this nomenclature is better.

### CODE REVIEW

**Comment by robrix** [comment]
It’s not just a conflict with `mtl`/`transformers`, but also a potential point of confusion since `MonadTrans`’ `lift` does not work like this `lift`. Would like to think about this a little further before we merge.

**Comment by patrickt** [comment]
Very fair. How about `raise`, `hoist`, or `promote`?

Interestingly enough, `lift` can be expressed as an `MFunctor` (from the mmorph package), which means that `Eff` is a monad in the category of monads… I think?

File: src/Control/Effect/Lift.hs [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
 import Control.Effect.Lift.Internal
 
 -- | Extract a 'Lift'ed 'Monad'ic action from an effectful computation.
 runM :: Monad m => Eff (LiftC m) a -> m a
 runM = runLiftC . interpret
 
+-- | Given a @Lift n@ constraint in a signature carried by @m@, use 'lift' to
+-- promote arbitrary actions of type @n a@ to @m a@.
+lift :: (Member (Lift n) sig, Carrier sig m, Functor n, Applicative m) => n a -> m a
+lift = send . Lift . fmap pure
```
Comment (robrix) [inline src/Control/Effect/Lift.hs]: What do you think about `sendM`, by analogy with `runM`, above?

Comment (patrickt) [reply src/Control/Effect/Lift.hs]: I think this is a good name. I'll document its spiritual similarity with `lift` from MTL.

**Pull request merged**